library/lib/serve.js

```diff
@@ -140,24 +140,40 @@ function possibleDirectories(path) {
 
 function serveIndexWithRouting(req, res, file) {
   const envRequire = isProduction ? require : require('import-fresh')
+
   const routeTemplate = envRequire(file)
 
-  const routes = routeTemplate.routes
   const location = parsePath(req.url)
 
-  return Promise.resolve(routes)
-    .then(routes => (routes && routes.match(location, req)) || { status: 200, data: null })
-    .then(data => {
-      if (!routes || !routes.resolveIndex) return [data, routeTemplate]
-
-      const indexLocation = routes.resolveIndex(location, req)
-      if (!indexLocation) return [data, routeTemplate]
-
-      const indexPath = resolve(target, publicPathDir, indexLocation, indexWithRouting)
-      return [data, envRequire(indexPath)]
-    })
-    .then(([{ status, headers, data }, template]) =>
-      Promise.resolve(template({ location, data })).then(html => [status, headers, html])
-    )
-    .then(([ status, headers, html ]) => res.status(status).set(headers).send(html))
+  const [dataOrPromise, template] = getDataAndRouteTemplate(routeTemplate, { location, req })
+
+  if (dataOrPromise.then) {
+    const dataPromise = dataOrPromise
+    dataPromise
+      .then(({ status, headers, data }) => {
+        const html = template({ location, data })
+        res.status(status).set(headers).send(html)
+      })
+  } else {
+    const data = dataOrPromise
+    const html = template({ location, data })
+    res.status(data.status).set(data.headers).send(html)
+  }
+}
```

Comment:
Er zat een klein foutje in bij de `else`, de verkeerde data werd gerenderd. 

Dat gebeurt minder snel als je nog een extra functie aanmaakt:

```suggestion
  if (dataOrPromise.then)
    dataOrPromise.then(data => renderTemplate(template, location, data))
  else
    renderTemplate(template, location, dataOrPromise)
}

function renderTemplate(template, location, { status, headers, data }) {
  const html = template({ location, data })
  res.status(status).set(headers).send(html)
}
```